variables.tf

```diff
@@ -0,0 +1,218 @@
+}
+
+variable "jenkins_instance_name" {
+  description = "The name to assign to the Jenkins VM"
+  default     = "jenkins"
+}
+
+variable "jenkins_instance_machine_type" {
+  description = "The machine type to provision for Jenkins"
+  default     = "n1-standard-4"
+}
+
+variable "jenkins_boot_disk_source_image" {
+  description = "The name of the disk image to use as the boot disk for the Jenkins master"
+  default     = "bitnami-jenkins-2-138-2-0-linux-debian-9-x86-64"
+}
+
+variable "jenkins_boot_disk_source_image_project" {
+  description = "The project within which the disk image to use as the Jenkins master boot disk exists"
+  default     = "bitnami-launchpad"
+}
+
+variable "jenkins_instance_zone" {
+  description = "The zone to deploy the Jenkins VM in"
+}
+
+variable "jenkins_instance_network" {
+  description = "The GCP network to deploy the Jenkins VM in"
+}
+
+variable "jenkins_instance_subnetwork" {
+  description = "The GCP subnetwork to deploy the Jenkins VM in"
+}
+
+variable "jenkins_instance_additional_tags" {
+  type        = "list"
+  description = "Additional tags to assign to the Jenkins VM"
+  default     = []
+}
+
+variable "jenkins_instance_additional_metadata" {
+  type        = "map"
+  description = "Additional instance metadata to assign to the Jenkins VM"
+  default     = {}
+}
+
+variable "jenkins_instance_network_tag" {
+  description = "The name of the tag to associate with the Jenkins instance. Useful if the default is already used in your GCP project"
+  default     = "jenkins"
+}
+
+variable "jenkins_initial_password" {
+  description = "The initial password to protect Jenkins logins with. Defaults to a random 8-character alphanumeric string. This may not contain special characters."
+  default     = ""
+}
+
+variable "jenkins_instance_access_cidrs" {
+  type        = "list"
+  description = "CIDRs to allow to access Jenkins over HTTP(s)"
+  default     = ["0.0.0.0/0"]
+}
+
+variable "jenkins_service_account_name" {
+  description = "The name of the service account to create for Jenkins VM provisioning"
+  default     = "jenkins"
+}
+
+variable "jenkins_service_account_display_name" {
+  description = "The display name of the service account to create for Jenkins VM provisioning"
+  default     = "Jenkins"
+}
+
+variable "jenkins_workers_project_id" {
+  description = "The GCP project to deploy Jenkins workers within"
+}
+
+variable "jenkins_workers_instance_cap" {
+  description = "The maximum number of GCE instances to create as Jenkins workers"
+  default     = 1
+}
+
+variable "jenkins_workers_description" {
+  description = "A description of the Jenkins worker cloud to show in Jenkins"
+  default     = "Jenkins worker"
+}
+
+variable "jenkins_workers_name_prefix" {
+  description = "A prefix for the Jenkins workers instance names"
+  default     = "jenkins"
+}
+
+variable "jenkins_workers_region" {
+  description = "The name of the region into which to deploy Jenkins workers"
+  default     = "us-east4"
```

Comment:
Why do the workers have a default region if the master doesn't? We can probably discover this by looking at the provided subnet.